src/app/store/trips/reducer.ts

```diff
@@ -0,0 +1,105 @@
+import { isDevMode } from '@angular/core';
+import { createReducer, MetaReducer, on } from '@ngrx/store';
+import { Trip } from '../../models/trips';
+import {
+  createNewTrip,
+  createNewTripComplete,
+  createNewTripError,
+  deleteTrip,
+  deleteTripComplete,
+  deleteTripError,
+  getAllTrips,
+  getAllTripsComplete,
+  getAllTripsError,
+  setSelectedTripId,
+  updateTrip,
+  updateTripComplete,
+  updateTripError,
+} from './actions';
+
+export const tripsFeatureKey = 'Trips';
+
+export interface TripState {
+  trips: Trip[];
+  isLoading: boolean;
+  selectedTripId?: string;
+}
+
+const initialState: TripState = {
+  trips: [],
+  isLoading: false,
+};
+
+export const tripsReducers = createReducer(
+  initialState,
+  on(getAllTrips, state => ({
+    ...state,
+    isLoading: true,
+  })),
+  on(getAllTripsComplete, (state, { trips }) => ({
+    ...state,
+    isLoading: false,
+    trips,
+  })),
+  on(getAllTripsError, (state, { error }) => {
+    alert(`Error getting trips. Error message: ${error.message}`);
+    return {
+      ...state,
+      isLoading: false,
+    };
+  }),
+  on(createNewTrip, state => ({
+    ...state,
+    isLoading: true,
+  })),
+  on(createNewTripComplete, (state, { trip }) => ({
+    ...state,
+    isLoading: false,
+    trips: [...state.trips, trip],
+  })),
+  on(createNewTripError, (state, { error }) => {
+    alert(`Error creating new trip. Error message: ${error.message}`);
+    return {
+      ...state,
+      isLoading: false,
+    };
+  }),
+  on(updateTrip, state => ({
+    ...state,
+    isLoading: true,
+  })),
+  on(updateTripComplete, (state, { trip }) => ({
+    ...state,
+    isLoading: false,
+    trips: [...state.trips, trip],
+  })),
+  on(updateTripError, (state, { error }) => {
+    alert(`Error updating trip. Error message: ${error.message}`);
```

Comment:
As an alternative to this, you can have an action to set the loading state to whatever the prop is.
Then you dispatch that from the error handler in the effect and do the user notifying in the effect